pkg/controller/plan/migration.go

```diff
@@ -1481,52 +1481,54 @@ func (r *Migration) updateConversionProgress(vm *plan.VMStatus, step *plan.Step)
 		return
 	}
 
-	if pod != nil {
-		switch pod.Status.Phase {
-		case core.PodSucceeded:
-			step.MarkCompleted()
-			step.Progress.Completed = step.Progress.Total
-		case core.PodFailed:
-			step.MarkCompleted()
-			step.AddError("Guest conversion failed. See pod logs for details.")
-		default:
-			el9, el9Err := r.Context.Plan.VSphereUsesEl9VirtV2v()
-			if el9Err != nil {
-				err = el9Err
-				return
-			}
-			if el9 {
-				err = r.updateConversionProgressEl9(pod, step)
-				if err != nil {
-					// Just log it. Missing progress is not fatal.
-					log.Error(err, "Failed to update conversion progress")
-					err = nil
-					return
-				}
-			}
-		}
-	} else {
+	if pod == nil {
 		step.MarkCompleted()
 		step.AddError("Guest conversion pod not found")
+		return
+	}
+
+	switch pod.Status.Phase {
+	case core.PodSucceeded:
+		step.MarkCompleted()
+		step.Progress.Completed = step.Progress.Total
+	case core.PodFailed:
+		step.MarkCompleted()
+		step.AddError("Guest conversion failed. See pod logs for details.")
+	default:
+		if pod.Status.PodIP == "" {
+			// we get the progress from the pod and we cannot connect to the pod without PodIP
+			break
+		}
+
+		el9, el9Err := r.Context.Plan.VSphereUsesEl9VirtV2v()
+		switch {
```

Comment:
i don't see this switch making it any easier. it was simpler in my opinion with a regular `if el9Err != nil`

Reply:
the idea is to say that we switch between different states of el9 and el9Err that are returned from `VSphereUsesEl9VirtV2v`
I think what made it more complicated for you is that first `el9Err == nil && el9` was checked and then `el9Err != nil` while the former can be simplified if it's checked after `el9Err != nil` - what do you think of the last changes?

Reply:
that's much better, thanks.